nixos/modules/services/networking/jibri/default.nix

```diff
@@ -0,0 +1,334 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.jibri;
+
+  # Copied from the jitsi-videobridge.nix file.
+  toHOCON = x:
+    if isAttrs x && x ? __hocon_envvar then ("\${" + x.__hocon_envvar + "}")
+    else if isAttrs x then "{${ concatStringsSep "," (mapAttrsToList (k: v: ''"${k}":${toHOCON v}'') x) }}"
+    else if isList x then "[${ concatMapStringsSep "," toHOCON x }]"
+    else builtins.toJSON x;
+
+  # We're passing passwords in environment variables that have names generated
+  # from an attribute name, which may not be a valid bash identifier.
+  toVarName = s: "XMPP_PASSWORD_" + stringAsChars (c: if builtins.match "[A-Za-z0-9]" c != null then c else "_") s;
+
+  defaultJibriConfig = {
+    id = "";
+    single-use-mode = false;
+
+    api = {
+      http.external-api-port = 2222;
+      http.internal-api-port = 3333;
+
+      xmpp.environments = flip mapAttrsToList cfg.xmppEnvironments (name: env: {
+        inherit name;
+
+        xmpp-server-hosts = env.xmppServerHosts;
+        xmpp-domain = env.xmppDomain;
+        control-muc = {
+          domain = env.control.muc.domain;
+          room-name = env.control.muc.roomName;
+          nickname = env.control.muc.nickname;
+        };
+
+        control-login = {
+          domain = env.control.login.domain;
+          username = env.control.login.username;
+          password.__hocon_envvar = toVarName "${name}_control";
+        };
+
+        call-login = {
+          domain = env.call.login.domain;
+          username = env.call.login.username;
+          password.__hocon_envvar = toVarName "${name}_call";
+        };
+
+        strip-from-room-domain = env.stripFromRoomDomain;
+        usage-timeout = env.usageTimeout;
+        trust-all-xmpp-certs = env.disableCertificateVerification;
+      });
+    };
+
+    recording = {
+      recordings-directory = "/tmp/recordings";
+      finalize-script = "/path/to/finalize"; # TODO(puck): replace with actual noop default
```

Comment:
You might want to fix this no be a proper noop script, or possibly remove the TODO? jibri's not clear on the status of `finalize-script`

Reply:
Thanks for your feedback! I'm working on an [option](https://github.com/cleeyv/nixpkgs/commit/b44c85ddff8a57890a9c63baab09398b03921adb) for the module to allow for configuring a finalize script, using the [example script](https://github.com/jitsi/jibri/blob/master/resources/finalize_recording.sh) from the jibri repo as a default. While trying to test this, I discovered a bug in this module's jibri instance that I've been stuck trying to figure out: https://github.com/jitsi/jibri/issues/441 Did you notice anything like this when using jibri for nixcon last year? I've considered trying to run an instance from your nixpkgs fork in nixcon-video-infra to compare to what I'm seeing in the updated version, but I haven't tried it yet.

Reply:
I don't believe we had this issue; I suspect there's something else obvious that's missing -- jibri's basically being `^C`'d by another process, and that's all your logs indicate (both error code 130 and the SIGINT point to that)-- try checking the full journal at the moment of finalizing; worst case cross-reference the `si_pid` with an output of `ps` :p